spec/javascripts/govuk_publishing_components/analytics/explicit-cross-domain-links.spec.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-env jasmine, jquery */
+/* global GOVUK */
+
+describe('Explicit cross-domain linker', function () {
+  'use strict'
+
+  var explicitCrossDomainLinks
+  var element
+  var linker
+
+  beforeEach(function () {
+    window.ga = {
+      getAll: function () {}
+    }
+
+    window.gaplugins = {
+      Linker: function () {}
+    }
+
+    linker = {
+      decorate: function () {}
+    }
+
+    explicitCrossDomainLinks = new GOVUK.Modules.ExplicitCrossDomainLinks()
+
+    spyOn(window.gaplugins, 'Linker').and.returnValue(linker)
+    spyOn(linker, 'decorate').and.returnValue('/somewhere?_ga=abc123')
+  })
+
+  afterEach(function () {
+    element.remove()
+    delete window.ga
+    delete window.gaplugins
+  })
+
+  describe('links', function () {
+    beforeEach(function () {
+      element = $('<a href="/somewhere">')
+    })
+
+    it('leaves the link href if unchanged there are no trackers in ga', function () {
+      spyOn(window.ga, 'getAll').and.returnValue([])
+      explicitCrossDomainLinks.start(element)
+      expect(element.attr('href')).toEqual('/somewhere')
+    })
+
+    it('modifies the link href to append ids from ga to the destination url', function () {
+      spyOn(window.ga, 'getAll').and.returnValue([{ ga_mock: 'foobar' }])
+      explicitCrossDomainLinks.start(element)
+      expect(element.attr('href')).toEqual('/somewhere?_ga=abc123')
```

Comment:
Is it worth testing that the ga param is appended correctly if there are already params on the href attribute?

Reply:
The test's stubbing the value to return (spyOn(linker, 'decorate').and.returnValue('/somewhere?_ga=abc123') above), not executing the actual ga linker code, so adding another test for a different url wouldn't test anything different. All this test is doing really is checking that linker.decorate gets called.

Reply:
Oh I see. Never mind then.